common/goos/Reader.cpp

```diff
@@ -142,6 +164,29 @@ Reader::Reader() {
   m_valid_source_text_chars[(int)'\n'] = true;
   m_valid_source_text_chars[(int)'\t'] = true;
   m_valid_source_text_chars[(int)'\r'] = true;
+
+  // allow every character that gets transformed to something else
+}
+
+bool Reader::is_valid_source_char(char c) {
+  if (!m_valid_source_chars_updated) {
+    for (auto& remap : g_font_large_char_remap) {
```

Comment:
`g_font_large_char_remap` isn't initialized when `pretty_printer_reader` (in `PrettyPrinter.cpp`) is initialized

Reply:
isn't the initialization just sorting the entries?  it should be fine to this before they are sorted?

Reply:
Oh I don't mean the sorting, I mean the values being set at all. You can see here they're not set before this specific `Reader` instance is created
![image](https://user-images.githubusercontent.com/7569514/137838933-f6964d21-4948-4b84-b6e1-6b8a7c8deaa9.png)

Reply:
ah yeah, I forgot that the pretty printer reader was also static. that's one of the problems with static stuff that needs initialization - the order is somewhat arbitrary: https://en.cppreference.com/w/cpp/language/siof

I've been trying to avoid having static stuff that's non-const or requires initialization in newer code because it tends to cause problems like this and with multi-threading (if we ever do that in the compiler...).  That said there are still some leftovers (pretty printer reader/symbol table, opcode table)